pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1836,14 +1836,23 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 
 // getCloudInitUserData : returns decrypted cloud-init user data
 func getCloudInitUserData(ctx *domainContext,
-	config types.DomainConfig) (*string, error) {
-	status, clearTextData, err := utils.GetCipherData(agentName,
-		config.CipherBlockStatus, config.CloudInitUserData)
-	publishCipherBlockStatus(ctx, status)
-	if status.IsCipher && len(status.Error) == 0 {
-		log.Infof("%s, cipherblock decryption successful\n", config.Key())
-	}
-	return clearTextData, err
+	dc types.DomainConfig) (zconfig.EncryptionBlock, error) {
+	if dc.CipherBlockStatus.IsCipher {
+		status, decBlock, err := utils.GetCipherCredentials(agentName,
+			dc.CipherBlockStatus)
+		ctx.pubCipherBlockStatus.Publish(status.Key(), status)
+		if err != nil {
+			log.Infof("%s, domain config cipherblock decryption unsuccessful: %v\n", dc.Key(), err)
+			decBlock.ProtectedUserData = *dc.CloudInitUserData
```

Comment:
Can we somehow check whether there was an encrypted user-data?
Then if there is no cleartext CloudInitUserData I think we should not fail the domain create, since we are not feeding it any fallback info. But I don't know if we can tell. Is some encrypted field zero length if zedcloud had nothing to encypt for user data?

Reply:
If there is nothing to encrypt, zedcloud will pass the nil cipher data in config. And if cipher data is nil then the IsCipher flag in cipher block status will not be set while parsing.